apps/api/src/app/endpoints/public/public.controller.ts

```diff
@@ -56,33 +61,77 @@ export class PublicController {
       hasDetails = user.subscription.type === 'Premium';
     }
 
+    const detailsPromise = this.portfolioService.getDetails({
+      impersonationId: access.userId,
+      userId: user.id,
+      withMarkets: true
+    });
+
+    const performance1dPromise = this.portfolioService.getPerformance({
+      dateRange: '1d',
+      impersonationId: undefined,
+      userId: user.id
+    });
+
+    const performanceMaxPromise = this.portfolioService.getPerformance({
+      dateRange: 'max',
+      impersonationId: undefined,
+      userId: user.id
+    });
+
+    const performanceYtdPromise = this.portfolioService.getPerformance({
+      dateRange: 'ytd',
+      impersonationId: undefined,
+      userId: user.id
+    });
+
+    const latestActivitiesPromise = this._orderService.getOrders({
+      includeDrafts: false,
+      take: 10,
+      sortColumn: 'date',
+      sortDirection: 'desc',
+      userCurrency:
+        this.request.user?.settings?.settings.baseCurrency ?? DEFAULT_CURRENCY,
```

Comment:
I think the currency of the user (not request) makes more sense:
`user.settings?.settings.baseCurrency ?? DEFAULT_CURRENCY,`